src/App.tsx

```diff
@@ -30,11 +30,32 @@ import AstarteProvider, { useAstarte } from './AstarteManager';
 import type { DashboardConfig } from './types';
 import Snackbar from './ui/Snackbar';
 import createReduxStore from './store';
+import { AstarteToken } from 'astarte-client';
 
 const DashboardSidebar = () => {
   const config = useConfig();
   const astarte = useAstarte();
 
+  type Session = {
+    realm: string;
+    token: string;
+    authUrl: string | null;
+  };
+
+  const claims = (() => {
+    let session: Session | null = null;
+    try {
+      session = JSON.parse(localStorage.getItem('session') || '');
+    } catch {
+      session = null;
+    }
+    if (session) {
+      const token = new AstarteToken(session.token);
+      return token.getClaims();
+    }
+    return null;
+  })();
+
```

Comment:
We are already accessing the Astarte React context which contains info about the current auth session:

```ts
  const astarte = useAstarte();
```

I suggest we expose the `token` in addition to the `realm` from `useAstarte`, so that we can use it like this:

```ts
  if (!astarte.isAuthenticated) {
    return null;
  }

  return (
    ...
    {astarte.token.can("realmManagement", "GET", "/interfaces") && <Sidebar.Item label="Interfaces" link="/interfaces" icon="interfaces" />}
    ...
    {astarte.token.can("appEngine", "GET", "/devices") && <Sidebar.Item label="Devices" link="/devices" icon="devices" />}
    {astarte.token.can("appEngine", "GET", "/groups") && <Sidebar.Item label="Groups" link="/groups" icon="groups" />}

    ...
  );.
```

There is no need to manually read and parse the session from localStorage. Same thing in the other files.